napari/utils/lock.py

```diff
@@ -0,0 +1,215 @@
+"""
+A generic lock class to be used as a locking mechanism.
+"""
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, validator
+
+from napari.utils.translations import trans
+
+
+class LockMode(Enum):
+    """Lock modes"""
+
+    EXACT = 0
+    IN_LIST = 1
+    IN_RANGE = 2
+    LARGER_THAN = 3
+    SMALLER_THAN = 4
```

Comment:
I'm not a huge user of enums, can we have values be "exact", "in_list", "in_range",... for easier debugability ? Or is there a perf reason?

Reply:
Actually, enums are slow in python. For 3.9, string equality test is faster (at least in my quick test)! I remember reading that they will be improved in 3.10 but I have not checked or done any tests. That being said, they are both very fast.

I am in the other side of the debate, but I do tend to complicate my codes with abstractions and so on (my favorite language is C# even though I don’t code in it that often). I prefer enums over string for the following reasons:
1. Enums can be auto completed while coding so errors can be caught easily even before running the code. For new developer, they can quickly explore the options.
2. I am guessing that enums will gain a substantial improvement in terms of performance over strings once python is compiled. I see this is the direction of Python for improving performance (e.g., mypyc).
3. For larger projects, enums help in the organization of code too. Other packages are starting to use enums too (pyside6 compared to pyside2)
4. I don’t expect this function will not be used by the napari “users” (it will be mainly used by napari developers and plugin developers) so it will not add more typing for users.

**That being said, does napari have any guidelines on which one to use? napari should have consistent codebase (whether being it string or enum). I am fine with both.**

Reply:
We use enums liberally for this sort of thing.  Let's stick with enums for now

Reply:
ok, will keep as enum.